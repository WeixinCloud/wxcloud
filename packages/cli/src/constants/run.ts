export const STAGE_TEXT: Record<string, string> = {
  BuildImage: '构建镜像',
  CreateCbrResource: '创建云托管资源',
  CreateEnv: '创建环境',
  CreateMysql: '创建数据库',
  CreateServer: '创建服务',
  CreateVersion: '创建版本',
  ExecuteSQL: '初始化数据库',
  ReleaseVersion: '发布版本',
  GrayRelease: '灰度测试',
  RunPipeline: '执行流水线'
};

export const STATUS_TEXT: Record<string, string> = {
  notInvolve: '已跳过',
  running: '进行中',
  failed: '失败',
  todo: '等待',
  finished: '完成',
  stopped: '已取消',
  unknown: ''
};

export const STAGE_COST: Record<string, string> = {
  BuildImage: '3 分钟',
  CreateCbrResource: '1 分钟',
  CreateEnv: '1 分钟',
  CreateMysql: '1 分钟',
  CreateServer: '1 分钟',
  CreateVersion: '1 分钟',
  ExecuteSQL: '1 分钟',
  ReleaseVersion: '1 分钟',
  RunPipeline: '2 分钟'
};
