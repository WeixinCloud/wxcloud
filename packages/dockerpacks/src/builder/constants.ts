export const WXCLOUDRUN_FILES_DIR = 'wxcloudrun-files';
