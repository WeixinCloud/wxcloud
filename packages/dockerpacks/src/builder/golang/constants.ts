export const GO_MOD = 'go.mod';
export const GO_SUM = 'go.sum';
export const VENDOR_DIR = 'vendor';
