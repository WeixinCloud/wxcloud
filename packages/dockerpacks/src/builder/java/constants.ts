export const SETTINGS_XML = 'settings.xml';
export const POM_XML = 'pom.xml';
export const MVN_EXTENSIONS_XML = '.mvn/extensions.xml';
export const BUILD_GRADLE = 'build.gradle';
export const BUILD_GRADLE_KTS = 'build.gradle.kts';
export const MANIFEST_MF = 'META-INF/MANIFEST.MF';
export const MAIN_CLASS = 'Main-Class';
