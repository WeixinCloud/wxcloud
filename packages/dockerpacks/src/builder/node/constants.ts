export const NODE_ENV_PRODUCTION = 'production';
export const PACKAGE_JSON = 'package.json';
export const NPM_SHRINKWRAP_JSON = 'npm-shrinkwrap.json';
export const PACKAGE_LOCK_JSON = 'package-lock.json';
export const YARN_LOCK = 'yarn.lock';
