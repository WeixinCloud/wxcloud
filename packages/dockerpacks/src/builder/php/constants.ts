export const COMPOSER_JSON = 'composer.json';
export const VENDOR_PATH = 'vendor';
export const NGINX_CONF_PATH = '/etc/nginx/nginx.conf';
