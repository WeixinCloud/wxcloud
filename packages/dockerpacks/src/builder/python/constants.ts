export const REQUIREMENTS_TXT = 'requirements.txt';
export const MANAGE_PY = 'manage.py';
