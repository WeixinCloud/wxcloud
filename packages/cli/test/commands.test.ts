import { test } from '@oclif/test';
import { TEST_APP_ID, TEST_APP_KEY, TEST_ENV_ID, TEST_SERVICE_ID } from './constants';
import { expect } from 'chai';

describe('login', () => {
  test
    .stdout()
    .command(['login', '-a', 'bad_id', '-k', 'bad_key'])
    .exit(201)
    .it('should fail to login with a bad key');

  test
    .stdout()
    .command(['login', '-a', TEST_APP_ID, '-k', TEST_APP_KEY])
    .it('should login successfully with a correct key');
});

describe('env', () => {
  describe('list', () => {
    test
      .stdout()
      .command(['env:list', '--json'])
      .it('should return correct environments list', ctx => {
        // @ts-ignore
        expect(ctx.stdout).toMatchSnapshot();
      });
  });
});

describe('service', () => {
  describe('list', () => {
    test
      .stdout()
      .command(['service:list', '-e', TEST_ENV_ID, '--json'])
      .it('should return correct services list', ctx => {
        // @ts-ignore
        expect(ctx.stdout).toMatchSnapshot();
      });
  });

  describe('create', () => {
    test
      .stdout()
      .command(['service:create', '-e', TEST_ENV_ID, '-s', TEST_SERVICE_ID, '--json'])
      .it('should create service successfully');
  });

  describe('config', () => {
    test
      .stdout()
      .command([
        'service:config',
        '-e',
        TEST_ENV_ID,
        '-s',
        TEST_SERVICE_ID,
        '-p',
        'foo=bar',
        '--noConfirm'
      ])
      .it('should update service config successfully');
  });
});

describe('logout', () => {
  test.stdout().command(['logout']).it('should logout successfully');
});
