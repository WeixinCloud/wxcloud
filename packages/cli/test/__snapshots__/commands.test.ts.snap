// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`env list should return correct environments list 1`] = `
"Wxcloud CLI 2.0.8
{\\"code\\":0,\\"errmsg\\":\\"success\\",\\"data\\":[{\\"Alias\\":\\"test-fisher\\",\\"EnvId\\":\\"test-fisher-8g4lxfcmc4b54f0f\\",\\"CreateTime\\":\\"2021-08-30 19:35:17\\"},{\\"Alias\\":\\"test-haotian\\",\\"EnvId\\":\\"test-haotian-6g03cruyb4101d37\\",\\"CreateTime\\":\\"2022-03-08 21:10:38\\"},{\\"Alias\\":\\"test-nanking\\",\\"EnvId\\":\\"test-nanking-2giy72uye67de253\\",\\"CreateTime\\":\\"2022-04-22 15:40:00\\"},{\\"Alias\\":\\"test-ning\\",\\"EnvId\\":\\"test-ning-5g1axa6h4a3b5fe6\\",\\"CreateTime\\":\\"2022-04-24 14:52:44\\"},{\\"Alias\\":\\"test-darky\\",\\"EnvId\\":\\"test-darky-3g3lrhgqc1817213\\",\\"CreateTime\\":\\"2022-05-09 11:19:26\\"},{\\"Alias\\":\\"test-fisher-standalo\\",\\"EnvId\\":\\"test-fisher-standalo-6b1b22ca3a8\\",\\"CreateTime\\":\\"2022-05-31 10:59:39\\"},{\\"Alias\\":\\"test-goreliu\\",\\"EnvId\\":\\"test-goreliu-1gy3wqrp6e2a1304\\",\\"CreateTime\\":\\"2022-05-31 11:07:38\\"},{\\"Alias\\":\\"test-fisher-main\\",\\"EnvId\\":\\"test-fisher-main-1f7gmsob46aec57\\",\\"CreateTime\\":\\"2022-05-31 15:36:11\\"},{\\"Alias\\":\\"nanatest\\",\\"EnvId\\":\\"nanatest-9gbwi4esb117c59d\\",\\"CreateTime\\":\\"2022-05-31 20:14:12\\"},{\\"Alias\\":\\"switch-backend\\",\\"EnvId\\":\\"switch-backend-1gos6kip81692379\\",\\"CreateTime\\":\\"2022-06-07 16:40:02\\"},{\\"Alias\\":\\"test-hotjar\\",\\"EnvId\\":\\"test-hotjar-2gxy58dre73b58a0\\",\\"CreateTime\\":\\"2022-06-08 13:26:13\\"},{\\"Alias\\":\\"standalone\\",\\"EnvId\\":\\"standalone-0g26eobz39db6ec9\\",\\"CreateTime\\":\\"2022-06-08 19:08:20\\"},{\\"Alias\\":\\"test-leogorliu\\",\\"EnvId\\":\\"test-leogorliu-2gobvr73b6c7c12c\\",\\"CreateTime\\":\\"2022-06-09 10:48:43\\"}]}
"
`;
